src/ORM/Query.php

```diff
@@ -1285,6 +1300,20 @@ public function insert(array $columns, array $types = [])
         return parent::insert($columns, $types);
     }
 
+    /**
+     * Returns a new Query that has automatic field aliasing disabled.
+     *
+     * @param \Cake\ORM\Table $table The table this query is starting on
+     * @return static
+     */
+    public static function subquery(Table $table)
+    {
+        $query = new static($table->getConnection(), $table);
```

Comment:
I think always using the `default` connection in `Table` is a design limitation that prevents read/write pool configurations. We should consider allowing read/write connection defaults in the future.